app/assets/javascripts/govuk_publishing_components/components/intervention.js

```diff
@@ -0,0 +1,80 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function Intervention ($module) {
+    this.$module = $module
+    this.$banner = this.$module.querySelector('.gem-c-intervention')
+    this.$closeLink = this.$module.querySelector('.gem-c-intervention__dismiss-link')
+    this.$campaignName = this.$module.getAttribute('data-intervention-name')
+    this.$interventionCookie = window.GOVUK.cookie('intervention_campaign')
+  }
+
+  Intervention.prototype.init = function () {
+    this.$module.close = this.handleClose.bind(this)
+
+    if (this.$closeLink) {
+      this.$closeLink.addEventListener('click', this.$module.close)
+    }
+
+    if (this.cookieHasCampaign()) {
+      this.hideBanner()
+    }
+  }
+
+  Intervention.prototype.handleClose = function (event) {
+    if (event) {
+      event.preventDefault()
+    }
+
+    if (this.cookieHasCampaign()) {
+      this.setCookies()
+    }
+    this.hideBanner()
+  }
+
+  Intervention.prototype.cookieHasCampaign = function () {
```

Comment:
This function is called four times in this file - would it be possible to call it once at the start, store the value, then refer to that for the rest?